### PR TITLE
Iae 14980 No Results

### PR DESCRIPTION
**Pull request by thomastighe**

### CODE REVIEW

File: src/ui-kit/experimental/hierarchical/hierarchical-tree-grid/hierarchical-tree-grid.component.html [hunk 1]
```diff
@@ -1,7 +1,12 @@
-<div class="example-container usa-grid">
-  <ng-container *ngTemplateOutlet="gridTemplate"></ng-container>
+<div class="example-container usa-grid" >
+  <ng-container *ngTemplateOutlet="gridData.length===0?noResults: gridTemplate"></ng-container>
```
Comment (colinalford-gsa) [inline src/ui-kit/experimental/hierarchical/hierarchical-tree-grid/hierarchical-tree-grid.component.html]: @thomastighe Can you add some whitespace between your operators for readability?

File: src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts [hunk 2]
```diff
@@ -127,16 +124,58 @@ export class HierarchicalDataService implements SamHiercarchicalServiceInterface
...
+    return data.sort((a, b) => {
+      let propertyA = this.sortingDataAccessor(a, sort.active);
+      let propertyB = this.sortingDataAccessor(b, sort.active)
+      const valueA = isNaN(+propertyA) ? propertyA : +propertyA;
+      const valueB = isNaN(+propertyB) ? propertyB : +propertyB;
+      return (valueA < valueB ? -1 : 1) * (sort.direction === 'asc' ? 1 : -1);
+    });
   }
 
+  private sortingDataAccessor: ((data: any, sortHeaderId: string) => string | number) =
```
Comment (colinalford-gsa) [inline src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts]: @thomastighe Isn't the initial type redundant? Couldn't you remote everything before the `=` and then remove the `=>` and just use it as a method?

File: src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts [hunk 3]
```diff
@@ -127,16 +124,58 @@ export class HierarchicalDataService implements SamHiercarchicalServiceInterface
   }
 
   getHiercarchicalById(item: SamHiercarchicalServiceSearchItem): Observable<SamHiercarchicalServiceResult> {
```
Comment (colinalford-gsa) [inline src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts]: @thomastighe Can you split this into a couple function calls for clarity and maintainability down the road? It almost looks like you could have a separate, small class for some of the logic in here.

Comment (thomastighe) [reply src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts]: Cleaned up and split to a new file and reused

File: src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts (hunk 2 above)
Comment (thomastighe) [reply src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts]: Cleaned this up

File: src/ui-kit/experimental/hierarchical/hierarchical-tree-grid/hierarchical-tree-grid.component.html (hunk 1 above)
Comment (thomastighe) [reply src/ui-kit/experimental/hierarchical/hierarchical-tree-grid/hierarchical-tree-grid.component.html]: added some spaces

**Pull request merged**